pkg/secrets/clients/conjur/mocks/conjur_secrets_retriever.go

```diff
@@ -11,24 +11,45 @@ import (
 	aren't concerned with if some Conjur secrets have permissions and others don't. Our main priority is
 	validating that those secrets with 'execute' permissions can be fetched.
 */
-var CanExecuteConjurVar bool
 
-func RetrieveConjurSecrets(_ []byte, variableIDs []string) (map[string][]byte, error) {
+type ConjurMockClient struct {
+	CanExecute bool
+	// TODO: CanExecute is really just used to assert on the presence of errors
+	// 	and should probably just be an optional error.
+	Database   map[string]string
+}
+
+func (c ConjurMockClient) RetrieveSecrets (_ []byte, variableIDs []string) (map[string][]byte, error) {
 	conjurSecrets := make(map[string][]byte)
 
-	if !CanExecuteConjurVar {
+	if !c.CanExecute {
 		return nil, errors.New("custom error")
 	}
 
 	for _, variableId := range variableIDs {
 		// Check if the secret exists in the mock Conjur DB
-		if _, ok := MockConjurDB[variableId]; !ok {
+		variableData, ok := c.Database[variableId]
+		if !ok {
 			return nil, errors.New("no_conjur_secret_error")
 		}
 
 		fullVariableId := fmt.Sprintf("account:variable:%s", variableId)
-		conjurSecrets[fullVariableId] = MockConjurDB[variableId]
+		conjurSecrets[fullVariableId] = []byte(variableData)
 	}
 
 	return conjurSecrets, nil
 }
+
+func NewConjurMockClient() ConjurMockClient {
+	database := map[string]string{
+		"conjur_variable1": "conjur_secret1",
```

Comment:
I see this uses hardcoded, "canned" database entries. Should we add a TODO to make the database entries configurable for a new mock client object? (e.g. by either adding a `map[]string` argument to `NewConjurMockClient`, or create an `AddSecret` method for the object)?

Reply:
I added an AddSecret()  in [363](https://github.com/cyberark/secrets-provider-for-k8s/pull/363/commits/6e2c10d9ffdf11475e20f1d9940976797fc64646